src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx

```diff
@@ -164,4 +166,84 @@ describe('Compose form component', () => {
     expect(draftMessageHeadingText).to.exist;
     expect(deleteButton).to.exist;
   });
+
+  it('displays user signature on /new-message when signature is enabled', async () => {
+    const customState = {
+      sm: {
+        triageTeams: { triageTeams },
+        categories: { categories },
+        draftDetails: {},
+        preferences: signatureReducers.signatureEnabled,
+      },
+    };
+
+    const screen = renderWithStoreAndRouter(
+      <ComposeForm recipients={triageTeams} draft={{}} />,
+      {
+        initialState: customState,
+        reducers: reducer,
+        path: Paths.COMPOSE,
+      },
+    );
+
+    const messageInput = await screen.getByTestId('message-body-field');
+
+    expect(messageInput)
+      .to.have.attribute('value')
+      .equal(
+        messageSignatureFormatter(signatureReducers.signatureEnabled.signature),
+      );
+  });
+
+  it('does not display user signature on /new-message when signature is disabled', async () => {
+    const customState = {
+      sm: {
+        triageTeams: { triageTeams },
+        categories: { categories },
+        draftDetails: {},
+        preferences: signatureReducers.signatureDisabled,
+      },
+    };
+
+    const screen = renderWithStoreAndRouter(
+      <ComposeForm recipients={triageTeams} draft={{}} />,
+      {
+        initialState: customState,
+        reducers: reducer,
+        path: Paths.COMPOSE,
+      },
+    );
+
+    const messageInput = await screen.getByTestId('message-body-field');
+
+    expect(messageInput).to.not.have.attribute('value');
+  });
+
+  it('does not append an existing draft message body with enabled signature', async () => {
+    const customState = {
+      sm: {
+        triageTeams: { triageTeams },
+        categories: { categories },
+        draftDetails: { draftMessage },
+        preferences: signatureReducers.signatureEnabled,
+      },
+    };
+
+    const screen = renderWithStoreAndRouter(
+      <ComposeForm recipients={triageTeams} draft={draftMessage} />,
+      {
+        initialState: customState,
+        reducers: reducer,
+        path: `/draft/7171715`,
```

Comment:
If you wanted to you could pull the id directly from draftMessage since I think we're trying to incorporate using response data more often. It's personal preference, but it looks good either way!

Reply:
great point. I would rather do that